src/lib/Lib/CMakeExecution.cpp

```diff
@@ -0,0 +1,164 @@
+//
+
+#include "lib/Lib/CMakeExecution.hpp"
+
+#include <llvm/Support/FileSystem.h>
+#include <llvm/Support/MemoryBuffer.h>
+#include <llvm/Support/Path.h>
+#include <llvm/Support/Program.h>
+
+namespace clang {
+namespace mrdocs {
+
+namespace {
+
+Expected<std::string>
+getCmakePath() {
+    auto const path = llvm::sys::findProgramByName("cmake");
+    MRDOCS_CHECK(path, "CMake executable not found");
+    std::optional<llvm::StringRef> const redirects[] = {llvm::StringRef(), llvm::StringRef(), llvm::StringRef()};
+    std::vector<llvm::StringRef> const args = {*path, "--version"};
+    int const result = llvm::sys::ExecuteAndWait(*path, args, std::nullopt, redirects);
+    MRDOCS_CHECK(result == 0, "CMake execution failed");
+    return *path;
+}
+
+Expected<std::string>
+executeCmakeHelp(llvm::StringRef cmakePath)
+{
+    llvm::SmallString<128> outputPath;
+    MRDOCS_CHECK(!llvm::sys::fs::createTemporaryFile("cmake-help", "txt", outputPath), 
+        "Failed to create temporary file");
+    std::optional<llvm::StringRef> const redirects[] = {llvm::StringRef(), outputPath.str(), llvm::StringRef()};
+    std::vector<llvm::StringRef> const args = {cmakePath, "--help"};
+    llvm::ArrayRef<llvm::StringRef> emptyEnv;
+    int const result = llvm::sys::ExecuteAndWait(cmakePath, args, emptyEnv, redirects);
+    MRDOCS_CHECK(result == 0, "CMake execution failed");
+
+    auto const bufferOrError = llvm::MemoryBuffer::getFile(outputPath);
+    MRDOCS_CHECK(bufferOrError, "Failed to read CMake help output");
+
+    return bufferOrError.get()->getBuffer().str();
+}
+
+Expected<std::string>
+getCmakeDefaultGenerator(llvm::StringRef cmakePath) 
+{
+    MRDOCS_TRY(auto const cmakeHelp, executeCmakeHelp(cmakePath));
+
+    std::istringstream stream(cmakeHelp);
+    std::string line;
+    std::string defaultGenerator;
+
+    while (std::getline(stream, line)) {
+        if (line[0] == '*' && line[1] == ' ') {
+            size_t const start = 2;
+            size_t const end = line.find("=", start);
+            if (end == std::string::npos) {
+                continue;
+            }
+            return line.substr(start, end - start);
+        }
+    }
+    return Unexpected(Error("Default CMake generator not found"));
+}
+
+Expected<bool>
+cmakeDefaultGeneratorIsVisualStudio(llvm::StringRef cmakePath) 
+{
+    MRDOCS_TRY(auto const defaultGenerator, getCmakeDefaultGenerator(cmakePath));
+    return defaultGenerator.starts_with("Visual Studio");
+}
+
+std::vector<std::string>
+parseCmakeArgs(std::string const& cmakeArgsStr) 
+{
+    std::vector<std::string> parsedArgs;
+    std::string arg;
+    bool inQuotes = false;
+
+    for (char ch : cmakeArgsStr) 
+    {
+        if (ch == '"') 
+        {
+            inQuotes = !inQuotes;
+        } 
+        else if (std::isspace(ch) && !inQuotes) 
+        {
+            if ( ! arg.empty()) 
+            {
+                if (arg[0] == '-' && arg.size() == 2) 
+                    continue;
```

Comment:
Why put keys in the same parsedArg as values?

Reply:
Answered here: https://github.com/cppalliance/mrdocs/pull/532#discussion_r1447108981